### PR TITLE
platformdirs 2.5.2

### PR DESCRIPTION
**Pull request by skupr-anaconda**
Changelog: https://github.com/platformdirs/platformdirs/blob/2.5.2/CHANGES.rst
License: https://github.com/platformdirs/platformdirs/blob/2.5.2/LICENSE.txt
Requirements: https://github.com/platformdirs/platformdirs/blob/2.5.2/pyproject.toml

Actions:
1. Remove `noarch`
2. Skip `py<37`
3. Add `hatchling` backend in host
4. Fix `python`
5. Add `doc_source_url`

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -8,23 +8,21 @@ package:
 
 source:
   url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/platformdirs-{{ version }}.tar.gz
-  sha256: 367a5e80b3d04d2428ffa76d33f124cf11e8fff2acdaa9b43d545f5c7d661ef2
+  sha256: 58c8abb07dcb441e6ee4b11d8df0ac856038f944ab98b7be6b27b2a3c7feef19
 
 build:
   number: 0
-  noarch: python
+  skip: True  # [py<37]
```
Comment (katietz) [inline recipe/meta.yaml:15]: shouldn't this be py<36 ?

**Review by katietz — CHANGES_REQUESTED** [review]

File: recipe/meta.yaml (hunk 1 above)
Comment (skupr-anaconda) [reply recipe/meta.yaml:15]: Nope, see https://github.com/platformdirs/platformdirs/blob/60fc40c294d90f7687f779487b8c19679fb73255/pyproject.toml#L20

Comment (skupr-anaconda) [reply recipe/meta.yaml:15]: python 3.6 support was dropped in v2.4.1 https://github.com/platformdirs/platformdirs/blob/2.5.2/CHANGES.rst#platformdirs-241-2021-12-26

Comment (katietz) [reply recipe/meta.yaml:15]: I see, I was just referring to the old statement <36.  Thanks for clarification

**Review by katietz — APPROVED** [review]
lgtm, thanks

**Review by marcoesters — APPROVED** [review]
LGTM

**Review by varlackc — APPROVED** [review]
LGTM

**Pull request merged**